src/mon/MgrStatMonitor.cc

```diff
@@ -136,6 +178,7 @@ version_t MgrStatMonitor::get_trim_to()
 
 void MgrStatMonitor::on_active()
 {
+  update_logger();
```

Comment:
`update_logger()` is already guarding for the same thing. Adding the check here, while a bit more verbose, would be redundant at this point. Otherwise, maybe check it here and assert on the function?